guides/plugins/apps/payment.md

```diff
@@ -0,0 +1,266 @@
+            <description lang="de-DE">Diese Zahlungsmethode wird die Transaktion auf 'offen' belassen.</description>
+            <!-- No URL is provided. -->
+        </payment-method>
+    </payments>
+</manifest>
+```
+{% endcode %}
+
+## Synchronous payments
+
+Synchronous payments are the more simple kind of payments without any further interaction with the user.
+You have two different kind of options here.
+Depending on if you have defined a `pay-url`, you can choose to be informed about - and possibly process - a payment or not.
+
+If you would just like to define another payment method like _advanced payment_ or _collect on delivery_, where the owner of the shop will manually mark the payment as `paid` later, you can stop reading this guide here, because no communication with your app is required.
+The transaction will remain on the status `open`.
+
+On the other hand, if you define a `pay-url`, you have the option to respond on this request with a different transaction status, for example `authorize`, `paid`or `fail`.
+This is useful if you would like to add a payment provider which only need the information the user has already provided in the checkout process and no additional information is required.
+This could be - for example - a simple credit check for payment upon invoice.
+
+Below you can see an example for a simple answer from your app to mark a payment as authorized. You can see two methods, `checkSignature` and `sign`, for both of these see the section [Validation](./payment.md#validation). The provided status should be the transistion name for the transaction based on the previous status `open`.
+
+```php
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route("/sync/process", name="sync.process", methods={"POST"})
+ */
+public function processSynchronousPayment(Request $request): JsonResponse
+{
+      // more on this in section "Validation"
+      $this->checkSignature($request);
+
+      $content = \json_decode($request->getContent(), true);
+
+      // implement your logic here based on the information provided in $content
+      $response = [ 'status' => 'authorize' ];
+
+      // this returns a json encoded response with the `shopware-app-signature` in the header
+      return $this->sign($response, $content['source']['shopId']);
+}
+```
+
+Instead of a successful response, you can also provide a failed response with a message (which will be logged) like this:
+
+```php
+$response = [
+    'status' => 'fail',
+    'message' => 'The customer failed to pass the credit check.',
+];
+```
+
+{% hint style="warning" %}
+Keep in mind that just by providing a `message`, the payment will default to status `fail`.
+{% endhint %}
+
+## Asynchronous payments
+
+Asynchronous payments are more complicated and rely on interaction with the user and therefore a redirect to the payment provider.
```

Comment:
JFYI - these line breaks won't be shown in Gitbook, as they don't contain a blank line. Just mentioning, in case that's not intended.